static/app/views/settings/components/teamSelectWithRole.tsx

```diff
@@ -0,0 +1,237 @@
+
+  const handleAddTeam = (option: Item) => {
+    const team = teams.find(tm => tm.slug === option.value);
+    if (team) {
+      onAddTeam(team.slug);
+    }
+  };
+
+  const renderBody = () => {
+    if (Object.keys(teamRoles).length === 0) {
+      return <EmptyMessage>{t('No Teams assigned')}</EmptyMessage>;
+    }
+
+    const rows = [...teamRoles].sort((a, b) => (a.teamSlug > b.teamSlug ? 1 : -1));
+
+    return (
+      <React.Fragment>
+        {organization.features.includes('team-roles') && (
+          <RoleOverwritePanelAlert
+            orgRole={orgRole}
+            orgRoleList={orgRoleList}
+            teamRoleList={teamRoleList}
+          />
+        )}
+        {rows.map(teamMembership => {
+          const {teamSlug, role} = teamMembership;
+          const team = teams.find(tm => tm.slug === teamMembership.teamSlug);
+          if (!team) {
+            return <div key={teamSlug}>Error finding data for #{teamSlug}</div>;
+          }
+
+          return (
+            <TeamRow
+              disabled={disabled}
+              key={teamSlug}
+              team={team}
+              teamRole={role}
+              orgRole={orgRole}
+              onRemove={slug => onRemoveTeam(slug)}
+              onChangeRole={onChangeRole}
+            />
+          );
+        })}
+      </React.Fragment>
+    );
+  };
+
+  // Only show options that aren't selected in the dropdown
+  const teamRoleSlugs = teamRoles.map(r => r.teamSlug);
+  const remainingTeams = teams.filter(tm => !teamRoleSlugs.includes(tm.slug));
+  const options = remainingTeams.map((team, index) => ({
+    index,
+    value: team.slug,
+    searchKey: team.slug,
+    label: <DropdownTeamBadge avatarSize={18} team={team} />,
+  }));
+
+  return (
+    <Panel>
+      <PanelHeader hasButtons>
+        {t('Team')}
+        <DropdownAutoComplete
+          items={options}
+          busyItemsStillVisible={fetching}
+          onChange={debounce<(e: React.ChangeEvent<HTMLInputElement>) => void>(
+            e => onSearch(e.target.value),
+            DEFAULT_DEBOUNCE_DURATION
+          )}
+          onSelect={handleAddTeam}
+          emptyMessage={t('No teams')}
+          // menuHeader={menuHeader}
+          disabled={disabled}
+          alignMenu="right"
+        >
+          {({isOpen}) => (
+            <DropdownButton
+              aria-label={t('Add Team')}
+              isOpen={isOpen}
+              size="xs"
+              disabled={disabled}
+            >
+              {t('Add Team')}
+            </DropdownButton>
+          )}
+        </DropdownAutoComplete>
+      </PanelHeader>
+
+      <PanelBody>{loadingTeams ? <LoadingIndicator /> : renderBody()}</PanelBody>
+    </Panel>
+  );
+}
+
+type TeamRowProps = {
+  onChangeRole: Props['onChangeRole'];
+  onRemove: Props['onRemoveTeam'];
+  orgRole: OrgRole['id'];
+  team: Team;
+  teamRole: TeamRole['id'] | null;
+  disabled?: boolean;
+};
+
+const TeamRow = ({
```

Comment:
Can we break this into a separate file?

Reply:
It's not being reused elsewhere, so splitting out a 40-50 LoC component seems futile.